test/qa/test_connect.py

```diff
@@ -450,3 +452,25 @@ def test_status_connected(tech, proto, obfuscated):
         assert time_passed - 1 <= time_connected <= time_passed + 1
 
     disconnect_base_test()
+
+
+@pytest.mark.parametrize(("tech", "proto", "obfuscated"), lib.STANDARD_TECHNOLOGIES)
+@pytest.mark.flaky(reruns=2, reruns_delay=90)
+@timeout_decorator.timeout(40)
+def test_connect_to_virtual_server(tech, proto, obfuscated):
```

Comment:
might be good to have also a test for autoconnect, because there the logic changed and now it is successful set only if a server is found with the user settings.